### ISSUE TITLE
Game will not launch

### ISSUE THREAD

**Issue by afromanDAN** [07th-mod/matsuribayashi#32]
## Bug Information

**Describe the bug**: Game does not start. It gives me an error about unity mismatching in the globalgamemanagers file.

**To reproduce**: Launching the game. Happens everytime.

**Expected behavior**: Game should start up like normal.

## Install Information

**The game**: You should only report Higurashi Ch.8 related issues in this repository.

**The mod variant**: Full

**I installed the patch**: Using the Installer

**I bought the game at**: Mangagamer

**Mod Install Date**: 11/5/2020

## System Information

**My operating system is**: Arch Linux

**State anything "special" about your computer's setup**: No, standard GPT install of arch linux.

## Screenshots
![Screenshot_20201105_034748](https://user-images.githubusercontent.com/20655561/98225168-02b59f00-1f1a-11eb-98c0-1503d3d84279.png)

If you need help locating logs or writing a bug report, see this page: https://07th-mod.com/wiki/Higurashi/support/

Logs attached here:
[afromandanEP08logs.zip](https://github.com/07th-mod/matsuribayashi/files/5493401/afromandanEP08logs.zip)

**Comment by ItaloKnox** [comment]
I think we don't ship a globalgamemanagers file with the patch, so I'm unsure how that could happen.

Did you get it from the Answer Arcs bundle or as a single purchase? Can you try using [this UI](https://07th-mod.com/ui.php?chapter=matsuribayashi&os=unix&unity=5.6.7f1) in your game and see if it fixes anything?

**Comment by drojf** [comment]
Arrgh, the installer says:

`Unity Version: Read [5.6.7f1] from [/home/[USERNAME]/Games/HigurashiEp08retry/HigurashiEp08_Data/resources.assets]`

but then later it installs: `https://07th-mod.com/ui.php?chapter=matsuribayashi&os=unix&unity=2017.2.5`

I'll have to check what's going on there, that's wrong

**Comment by drojf** [comment]
I think this line was wrong in the `installData.json`:



`{"name": "ui",    "id": "ui-linux-old", "os": ["linux"], "unity": "5.6.7f1", "url":"https://07th-mod.com/ui.php?chapter=matsuribayashi&os=unix&unity=2017.2.5"},` [(link)](https://github.com/07th-mod/python-patcher/blob/ae6c985784ca2c5d77c8478854cacc4e007a8e01/installData.json#L608)

I'm guessing it should be the link you just posted, https://07th-mod.com/ui.php?chapter=matsuribayashi&os=unix&unity=5.6.7f1?

**Comment by ItaloKnox** [comment]
I don't think that's wrong, we have compatibility packs that were only for GOG that was stuck on the previous version. It should install the compatibility pack and then get the UI for the newest version.

It's a bit confusing to explain, @ooa113y might be the best person to know if it's wrong or not.

**Comment by ooa113y** [comment]
Yeah, this is correct. We basically supply the same UI as for the main game, and then the compatibility pack more or less contains a copy of the entire Steam game (except for assets) that is used to "upgrade" the outdated GOG version to the new version of Unity.

**Comment by drojf** [comment]
> I think we don't ship a globalgamemanagers file with the patch, so I'm unsure how that could happen.

There is a `globalgamemanagers` file in the comparability pack: https://07th-mod.com/rikachama/matsuri-system/Matsuribayashi-System_unix.7z

The log first says player data archive not found at (~home path)

Then it says "using local filesystem", which then loads the `globalgamemanagers` which came with the comparability pack.

However it says it expects `5.6.7f1`, and the one with the compatability pack is `2017.2.5f1` - I would have expected it to expect `2017.2.5f1`....so I'm not really sure what it is

**Comment by ItaloKnox** [comment]
Yeah, I just checked the packs and the file is there. Which is even weirder because the compatibility packs had a 100% success rate until now. Maybe the OP can try a manual installation, so we can rule out the possibility of the installer doing things in the wrong order?

**Comment by afromanDAN** [comment]
I just bought it as a single purchase. I redownloaded it from managagamer and reinstalled the mod and now I get an error like this.

![image](https://user-images.githubusercontent.com/20655561/98316735-95991c80-1fa0-11eb-8a42-3c121e2677b1.png)

**Comment by drojf** [comment]
Just a sanity check - the unmodded game works fine yea?

**Comment by afromanDAN** [comment]
Yes, the game plays fine.

**Comment by drojf** [comment]
I just installed ep8 from Mangagamer on Ubuntu and I'm getting a similar crash after patching the game on a fresh install - @afromanDAN don't worry about trying things out until we at least get it working on our end.

**Comment by afromanDAN** [comment]
Ah so I'm not crazy then lol.

**Comment by Shinkirou** [comment]
I'm also having the same issue with the mono SIGABRT crash on Mint 20. The unpatched game runs fine. I tried to update the mono packages but the crash persisted.

**Comment by drojf** [comment]
So I got my own install working, but I'm not sure if the sharedassets0.assets versions you guys have are the same as mine.

I'm also not sure if our installs are the same - it's best that you re-download the game from Mangagamer to make sure we have the same versions of Matsuribayashi.

Could you guys do the following:

1. Re-download Matsuribayashi from Mangagamer (or GOG). **Don't install the mod yet!**
    - @Shinkirou please tell me which store you bought the game from - Mangagamer or GOG
2. Check if your unmodded `Higurashi When They Cry Hou - Ch.8 Matsuribayashi/HigurashiEp08_Data/sharedassets0.assets` is identical to my stock one listed below (use `cmp <yoursharedassets> <mysharedassets>` or `diff` or compare checksums)
3. If they are identical:
    - install the mod 
    - once the mod is installed, copy over the `sharedassets0.assets` file with the modified one in the download link below (the >100mb one)
4. If they are not identical, upload your stock `sharedassets0.assets` so I have a copy of it/can modify it. In the upload, please note the store you downloaded the game from, and the date.

----

My Stock/unmodded Mangagamer `sharedassets0.assets` (2020-11-12) SHA256: A200EC2A85349BC03B59C8E2F106B99ED0CBAAA25FC50928BB8BA2E2AA90FCE9

Mangagamer `sharedassets0.assets` after modification (to be used with the mod) SHA256:
0FD3E6A38FE250B2E736B101BC3F64A723D6FC29A68BA55C9451D4621326E1F7 

OneDrive Download link for these two files: https://1drv.ms/u/s!Ar-lAVeetlqhgsZuO26dC2dz5he_7A?e=7pNa5L (if it asks you to sign in, just use a private/incognito window)

I think we have previously assumed that sharedassets0.assets with the same unity version and the same operating system are "the same", but it appears that the steam and mangagamer versions of the file are different, despite both having the same unity version and operating system.

**Comment by Shinkirou** [comment]
It worked after changing the modded sharedassets file.

I'm running the Mangagamer store version. The original sharedassets file was identical to the one you sent. After installing the mod and replacing the modded shareassets, the game was able to run without any issues.

Thanks for the quick help!

**Comment by afromanDAN** [comment]
My stock mangagamer sharedassets0.assets has the same SHA256. As with Shinkirou above your modified version fixed my game also.

**Comment by afromanDAN** [comment]
well I finished the whole game with out any issues with your new patch.

**Comment by drojf** [comment]
Thanks for reporting back afroman, at least I know this doesn't break anything.

Before I can put the new file in the installer, I need to update the installer so it can use the correct file automatically (the above issue 07th-mod/python-patcher#128), but hopefully that will fix it for new users.

**Comment by drojf** [comment]
We haven't had any reports of this issue after I applied the fix to the installer, so I think this issue can be closed now.

**Comment by runboy93** [comment]
Lubuntu 20.04.3 LTS

Matsuribayashi won't start up, I mean like when I click start.sh nothing happens.

Nothing appears on process manager (Htop) and so on very hard to say what is wrong.

**Comment by drojf** [comment]
Hi, please answer the following questions:

1. When did you download your copy of the game from GOG? Are you using an old version of the installer you downloaded a long time ago, or did you download it from GOG just recently (or use GOG galaxy to install?)

2. Did you install the mod manually or using the installer?

3. Can you please send the install log (if you installed using the installer)? It should be in your browser downloads as a .zip file.

----

If you installed manually, make sure you installed the compatibility pack for GOG Linux at the bottom of this page (since you have the Linux GOG release): https://07th-mod.com/wiki/Higurashi/Manual-Installation/

That is, if you installed manually, you need these files:
 - Compatibility pack: https://07th-mod.com/rikachama/matsuri-system/Matsuribayashi-System_unix.7z
 - UI file: https://07th-mod.com/rikachama/ui/Matsuribayashi-UI_2017.2.5_unix_gog_mg.7z
     - ONLY IF above UI file didn't work, try this UI file: https://07th-mod.com/ui.php?chapter=matsuribayashi&os=unix&unity=2017.2.5

**Comment by runboy93** [comment]
@drojf 

1) Downloaded game today from GoG (not using Galaxy, direct downloads) 07th mod installer v1.1.89 (latest for now)

2) Installer

3) update soon

There was earlier posts about SHA256 on their `sharedassets0.assets` files, mine (stock, unmodified) is:
`a1f7b39665d63d704bc14275dcdf71cf4cca9532ac61bee23b5b18b62a5cd3af`

**Comment by drojf** [comment]
@runboy93 did you do anything between your first install attempt and your second install attempt?

I can see that on your first install attempt, it detected that your `sharedassets0.assets` file was unknown, so it stopped the install. But it seems to have gone away on your second attemptd. Did you manually replace that file?

**Comment by drojf** [comment]
actually, the first install in that zip file is probably not your first install, because you already have a sharedassets0.assets.backup file there, which means the installer ran at least once before.

**Comment by runboy93** [comment]
So problem is with my assets file or what?

**Comment by drojf** [comment]
@runboy93 I did some testing and can reproduce the problem on my end too.

I think the installer is not applying the compatibility pack correctly.

Can you try to manually copy the contents of this archive into the `Higurashi When They Cry Hou Ch 8 Matsuribayashi/game` folder: https://07th-mod.com/rikachama/matsuri-system/Matsuribayashi-System_unix.7z

It should overwrite the `HigurashiEp08_Data` folder, and the game executables when you copy it.

**Comment by runboy93** [comment]
Thanks it does start up normally now!

**Comment by drojf** [comment]
Ah good to know.

----

For the developers, I've raised this issue on the installer repo: https://github.com/07th-mod/python-patcher/issues/162

This issue can be closed once it's verified that the installer can install to the GOG release successfully (once the installer is fixed).
